.github/workflows/push_tests.yml

```diff
@@ -81,3 +82,22 @@ jobs:
           valColorRange: ${{ env.COVERAGE_PCT }}
           maxColorRange: 95  # value where ligth green starts
           minColorRange: 50  # value where red starts
+
+  # Style checking
+  style_check:
+    name: Style and file check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
+      - name: Install dependencies
+        run: |
+          pip install pre-commit
+      - name: Style and file check
+        run: |
+          pre-commit run --all-files --verbose --show-diff-on-failure
+
+
```

Comment:
the name of the file `push_tests.yml` suggests that the steps defined here are running the tests on push.
maybe it makes sense to move `style_check` to a different file, say `push_formatting.yml`?

On the other hand this could be the last step of the job `tests`, so we simply add `pre-commit` to `environment.yaml` and avoid the duplication of checking out the repo and installing dependencies.

Reply:
If one of the tasks in the workflow fails, are the other tasks even carried out? If not, then I would suggest moving formatting to another workflow because it would be annoying not to see if the formatting is right if unit tests or sth else fails

Reply:
Just to clarify: if I am not using the pre commit hooks locally, and I try to push something, the job `style_check` will always fail because files are modified by the hook (or at least I got this impression from the actions `try workflow`).
Hence, each developer should (is forced to) install the pre commit hooks in order to make this test pass.

Reply:
@srosenbu Since it is a own job ("style check") you will see two workflows in the actions (you can have a look at my serveral tries :) ). So even if the style check fails you will see that the test are working and vis versa.

@pdiercks I think it is not necessary to install the pre-commit, if you are working with black and isort anywhy (e.g. in pycharm you can add those things as external tool and your code is checked automatically on save!)

Reply:
@aradermacher 

> @pdiercks I think it is not necessary to install the pre-commit, if you are working with black and isort anywhy (e.g. in pycharm you can add those things as external tool and your code is checked automatically on save!)

but that means I have to run black and isort on the files I want to commit (independent of how I might do that).
That was the question I was trying to ask 😄